ui/app/components/app/transaction-activity-log/transaction-activity-log.container.js

```diff
@@ -1,5 +1,5 @@
 import { connect } from 'react-redux'
-import R from 'ramda'
+import findLastIndex from 'lodash/findLastIndex'
```

Comment:
Lodash doesn't do tree shaking, you either have to install the individual modules (lodash-flatten) or import from these paths to only get the functions you want. I'm not sure how dated this knowledge is but checking bundlephobia.com for lodash shows it does not yet (or will never) support tree shaking.

Reply:
I don't think it matters for us; the entirety of `lodash` will be included regardless, because _many_ transitive dependencies use it directly in this manner.

Plus v5 of lodash (when/if it ever gets released) supposedly will support tree shaking... or that was the plan, so this format seemed more forward-facing 🤷‍♂️

Reply:
Have we ever done a dependency audit or bundle analyzer to see what the actual impact of that is for users? If v5 lands, and we upgrade, we'll be shipping multiple versions of lodash until every package in our dep tree upgrades. I get that we don't have the same network constraints as a normal web app given that our code is downloaded and installed rather than fetched over the network, but bundle size impacts perf as well. I'd be curious to see where we stand today.

Also, I am not gonna die on this hill so if you feel strongly it should be a destructured import I will change it.

Reply:
I didn't mean to suggest that we'd update to v5 right away. But the minute that v5 is included as a transitive dependency, it'll make no difference to the bundle size either way, so we'll get to update "for free".

And similarly, using the `ldoash/[method]` syntax today makes no impact, because our bundle has the entirety of `lodash` in it regardless today.

Reply:
You'll only get the benefit of v5 of lodash when *all* dependencies update to it or move it to peer dependency (if they aren't already). I'll change the import syntax as you suggested.

How are we measuring our bundle size currently and deriving from it which dependencies are "heavy"?

Reply:
I didn't mean to suggest we'd get any benefit from switching to v5 :P Just that once a dependency makes that decision for us, there wouldn't be any drawbacks either.

We aren't doing a great job of tracking changes in bundle size at the moment; that's something I had hoped to add to the MetaMask bot comment at some point. There is a dependency visualization generated with each build though, which is linked in that comment. It shows the size of each dependency.

Reply:
Oh dang, can't believe I never clicked the expander! Thanks, @Gudahtt. In retrospect, lodash is fine at being 2.8% of our total UI bundle. We don't use Material UI nearly enough for it to warrant being 5% of our bundle :P